self-hosting/README.md

```diff
@@ -0,0 +1,304 @@
+
+#### 64-bit Ubuntu
+
+- [Docker CE](https://docs.docker.com/engine/install/ubuntu/)  
+
+- [Docker-compose](https://docs.docker.com/compose/install/)
+
+### Windows 
+
+- [WSL2](https://docs.microsoft.com/en-us/windows/wsl/install-win10)
+
+- [Docker for Windows here](https://download.docker.com/win/stable/Docker%20for%20Windows%20Installer.exe)  and [here](https://docs.docker.com/docker-for-windows/install/)
+
+Note: If you have Hyper-V Capability, please ensure it is enabled in order to run Linux Containers on Windows. If you are running your Windows Server in cloud services, please ensure it is running on [bare-metal](https://en.wikipedia.org/wiki/Bare_machine). You will not be able to run Linux Containers in Windows if the previous comments are not adhered due to nested virtualization.
+
+If you do not have Hyper-V capability, but your server still supports virtualization, ensure that is enabled in your BiOS, and install the following package: [Docker Toolbox using VirtualBox](https://github.com/docker/toolbox/releases)
+
+### Mac OSX:
+
+- [Docker for Mac](https://download.docker.com/mac/stable/Docker.dmg)
+
+## Installing with a compose file
+
+The CHT containers are installed using [docker compose](https://docs.docker.com/compose/reference/overview/) so that you can run multiple containers  as a single service.
+
+Start by choosing the location where you would like to save your compose configuration file.  Then create your docker compose file (`docker-compose.yml`) with contents as listed below.
+
+```yaml
+version: '3.7'
+
+services:
+  medic-os:
+    container_name: medic-os
+    image: medicmobile/medic-os:cht-3.9.0-rc.2
+    volumes:
+      - medic-data:/srv
+    ports:
+     - 80:80
+     - 443:443
+    working_dir: /srv
+    depends_on:
+      - haproxy
+    networks:
+      - medic-net
+    environment:
+      - DOCKER_NETWORK_NAME=haproxy
+      - DOCKER_COUCHDB_ADMIN_PASSWORD=$DOCKER_COUCHDB_ADMIN_PASSWORD
+
+  haproxy:
+    container_name: haproxy
+    image: medicmobile/haproxy:rc-1.17
+    volumes:
+      - medic-data:/srv
+    environment:
+      - COUCHDB_HOST=medic-os
+      - HA_PASSWORD=$DOCKER_COUCHDB_ADMIN_PASSWORD
+    networks:
+      - medic-net
+
+volumes:
+  medic-data:
+    name: medic-data
+
+networks:
+  medic-net:
+    name: medic-net
+
+```
+
+The install requires an admin password that it will configure in the database. You need to provide this externally as an environment variable. before you run the compose file, you need to export this variable as shown below.
+
+`export DOCKER_COUCHDB_ADMIN_PASSWORD=myAwesomeCHTAdminPassword`
+
+You can then run docker-compose in the folder where you put your compose configuration yaml file as shown below.
+
+```bash
+## run compose inside the coconfigulation folder as root 
+
+$ sudo docker-compose up 
```

Comment:
[Style guide says](https://docs.communityhealthtoolkit.org/contribute/docs/style-guide/#dont-include-the-command-prompt) we don't include the `$` in code snippet.  There's 13 total instances of this below.  Gotta catch 'em all ;) (but I'm not a Pokémon fan).